.azure-pipelines/pipelines.yml

```diff
@@ -305,6 +286,43 @@ stages:
               artifactName: docker
             condition: always()
 
+  - stage: docs
+    dependsOn: ["docker"]
+    condition: and(succeeded(), eq(variables['PostSubmit'], true), ne(variables['NoSync'], true))
+    jobs:
+      - job: publish
```

Comment:
Can this depends on the artifacts uploaded by previous job?

Reply:
not sure if i follow exactly - do you mean the docker artifacts? that would be useful in terms of getting the image digests but will take quite a bit of time as the images are very large.

if you meant the previous `docs` job, that is excluded on master, so this is the only docs job

Reply:
No, just like what we do for Envoy binary, we do upload docs as BuildArtifact:

https://github.com/envoyproxy/envoy/blob/master/.azure-pipelines/pipelines.yml#L69-L72

And you can just use DownloadBuildArtifacts here instead of rebuilding, like https://github.com/envoyproxy/envoy/blob/master/.azure-pipelines/pipelines.yml#L272-L278

Reply:
there is a benefit to actually building the docs after docker.

if you want to link to the docker images you *must* know the image digest.

this cant be known until after the docker run has complete.

in a follow up, it was my intention to upload just the digests as an artifcact, so the docs build could read them  and create links

Reply:
ok that sounds reasonable.